src/index.js

```diff
@@ -0,0 +1,221 @@
+import { createServer } from 'http';
+import fs from 'fs/promises';
+import path from 'path';
+import { fileURLToPath } from 'url';
+import { parse as parseUrl } from 'url';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const PORT = process.env.PORT || 3000;
+
+// Simple static file server
+const mimeTypes = {
+  '.html': 'text/html',
+  '.js': 'text/javascript',
+  '.css': 'text/css',
+  '.json': 'application/json',
+  '.png': 'image/png',
+  '.jpg': 'image/jpg',
+  '.gif': 'image/gif',
+  '.svg': 'image/svg+xml',
+  '.ico': 'image/x-icon'
+};
+
+async function serveStaticFile(filePath, res) {
+  try {
+    const data = await fs.readFile(filePath);
+    const ext = path.extname(filePath);
+    const contentType = mimeTypes[ext] || 'application/octet-stream';
+    res.writeHead(200, { 'Content-Type': contentType });
+    res.end(data);
+  } catch (error) {
+    res.writeHead(404);
+    res.end('Not found');
+  }
+}
+
+const server = createServer(async (req, res) => {
+  const parsedUrl = parseUrl(req.url, true);
+  let pathname = parsedUrl.pathname;
+  
+  // Default to index.html
+  if (pathname === '/') {
+    pathname = '/index.html';
+  }
+  
+  // Serve from public directory with path traversal protection
+  const publicDir = path.join(__dirname, '../public');
+  const requestedPath = path.join(publicDir, pathname);
+  const resolvedPath = path.resolve(requestedPath);
+  
+  // Security: Ensure the resolved path is within the public directory
+  if (!resolvedPath.startsWith(path.resolve(publicDir))) {
+    res.writeHead(403);
+    res.end('Forbidden');
+    return;
+  }
+  
+  await serveStaticFile(resolvedPath, res);
+});
+
+// Load configuration
+let config;
+try {
+  const configData = await fs.readFile(path.join(__dirname, '../config/sections.json'), 'utf-8');
+  config = JSON.parse(configData);
+} catch (error) {
+  console.error('Error loading config:', error);
+  process.exit(1);
+}
+
+/**
+ * Call Ollama API to generate content
+ */
+async function callOllama(systemPrompt, userPrompt) {
+  const url = `${config.ollamaConfig.baseUrl}/api/generate`;
+  
+  try {
+    const response = await fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        model: config.ollamaConfig.model,
+        prompt: userPrompt,
+        system: systemPrompt,
```

Comment:
The system prompt should be a concatenation of a global config.systemPrompt and the intructions that's specific for this article